internal/federationout/federationout.go

```diff
@@ -86,153 +87,253 @@ func (s Server) Fetch(ctx context.Context, req *pb.FederationFetchRequest) (*pb.
 	return response, nil
 }
 
-func (s Server) fetch(ctx context.Context, req *pb.FederationFetchRequest, itFunc iterateExposuresFunc, fetchUntil time.Time) (*pb.FederationFetchResponse, error) {
+func (s Server) fetch(ctx context.Context, req *federation.FederationFetchRequest, itFunc iterateExposuresFunc, fetchUntil time.Time) (*federation.FederationFetchResponse, error) {
 	logger := logging.FromContext(ctx)
 	metrics := s.env.MetricsExporter(ctx)
 	metricsMiddleWare := metricsware.NewMiddleWare(&metrics)
 
-	for i := range req.RegionIdentifiers {
-		req.RegionIdentifiers[i] = strings.ToUpper(req.RegionIdentifiers[i])
+	if in := intersect(req.IncludeRegions, req.ExcludeRegions); len(in) > 0 {
+		logger.Errorw("overlap in included and excluded regions", "intersection", in)
+		return nil, fmt.Errorf("overlap in include and exclude region set: %v", in)
 	}
-	for i := range req.ExcludeRegionIdentifiers {
-		req.ExcludeRegionIdentifiers[i] = strings.ToUpper(req.ExcludeRegionIdentifiers[i])
+
+	for i, region := range req.IncludeRegions {
+		req.IncludeRegions[i] = strings.ToUpper(region)
+	}
```

Comment:
I think you sorted and trimmed space these regions earlier - do you need to sort/trimspace here?